File: score_extraction/score_extraction_functions.R
library(tidyverse)
library(stringr)
library(stringi)


contexto_score<-function(contexto_text){
  return(as.integer(str_extract(contexto_text, "(?<=in\\s)(.*?)(?=\\sguesses)")))
}

wordle_score<-function(wordle_text){
  return(as.integer(str_extract(wordle_text, "\\d+(?=/)")))
}

bandle_score<-function(bandle_text){
  return(as.integer(str_extract(bandle_text, "\\d+(?=/)")))
}

spotle_score<-function(spotle_text){
  return(nchar(sub("#\\d+", "", spotle_text)) - 21)
}

File: score_extraction/score_extraction_functions.js
// Load required libraries
const strExtract = (text, pattern) => {
    const match = text.match(new RegExp(pattern));
    return match ? parseInt(match[0]) : null;
};
 
const contextoScore = (contextoText) => {
    const pattern = '(?<=in\s)(.*?)(?=\sguesses)';
    const match = contextoText.match(new RegExp(pattern));
    return match ? parseInt(match[1]) : null;
};
 
const wordleScore = (wordleText) => {
    const pattern = '\\d+(?=/)';
    return strExtract(wordleText, pattern);
};
 
const bandleScore = (bandleText) => {
    const pattern = '\\d+(?=/)';
    return strExtract(bandleText, pattern);
};
 
const spotleScore = (spotleText) => {
    const pattern = '#\\d+';
    const cleanedText = spotleText.replace(new RegExp(pattern), '');
    return cleanedText.length - 21;
};
